src/net.cpp

```diff
@@ -1278,7 +1278,8 @@ void ThreadSocketHandler()
                     }
                     if (fDelete) {
                         vNodesDisconnected.remove(pnode);
-                        assert(std::find(vNodes.begin(),vNodes.end(), pnode) == vNodes.end());  // make sure it has been removed
+                        // no need to remove from vNodes. we know pnode has already been removed from vNodes since that
```

Comment:
Unless it is structurally enforced (and it looks like it isn't) an assert is still a good idea.